services/core-web/src/components/mine/ExplosivesPermit/MineExplosivesPermitTable.tsx

```diff
@@ -94,177 +104,767 @@ const MineExplosivesPermitTable: FC<RouteComponentProps & MineExplosivesPermitTa
     <EyeOutlined className="padding-sm icon-lg icon-svg-filter" />
   );
 
-  const columns: ColumnType<MineExplosivesTableItem>[] = [
-    {
-      title: "Permit #",
-      dataIndex: "permit_number",
-      render: (text) => (
-        <div title="Permit #" className={hideColumn(!isPermitTab)}>
-          {text}
-        </div>
-      ),
-      sorter: false,
-      className: hideColumn(!isPermitTab),
-    },
-    {
-      title: "Application #",
-      dataIndex: "application_number",
-      render: (text) => (
-        <div title="Application #" className={hideColumn(isPermitTab)}>
-          {text}
-        </div>
-      ),
-      sorter: false,
-      className: hideColumn(isPermitTab),
-    },
-    {
-      title: "Mines Act Permit #",
-      dataIndex: "mines_permit_number",
-      render: (text) => <div title="Mines Act Permit #">{text}</div>,
-      sorter: false,
-    },
-    {
-      title: "Notice of Work #",
-      dataIndex: "now_number",
-      render: (text) => <div title="Notice of Work #">{text || Strings.EMPTY_FIELD}</div>,
-      sorter: false,
-    },
+  const columns: ColumnType<MineExplosivesTableItem>[] = isFeatureEnabled(
+    Feature.ESUP_PERMIT_AMENDMENT
+  )
+    ? [
+        {
+          title: "ESUP #",
+          key: "permit_number",
+          render: (record) => {
+            return (
+              <Row justify="space-between">
+                <Col>
+                  <Typography.Text>{record.permit_number || Strings.EMPTY_FIELD}</Typography.Text>
+                </Col>
+                {record.explosives_permit_amendments.length > 1 && (
+                  <Col className="amendments-badge">
+                    <FontAwesomeIcon icon={faFiles} />
+                    {record.explosives_permit_amendments.length - 1}
+                  </Col>
+                )}
+              </Row>
+            );
+          },
+        },
+        {
+          title: "Mines Act Permit #",
+          dataIndex: "mines_permit_number",
+          key: "mines_permit_number",
+        },
+        {
+          title: "Notice of Work #",
+          dataIndex: "now_number",
+          key: "now_number",
+        },
+        {
+          title: "Amendments",
```

Comment:
This table wants more CoreTableCommonColumns functionality and fewer ternaries. Can we separate out the feature flag stuff even so it's easier to remove later? I am okay with defining multiple variables and deciding which one to use in an if statement or whatever.

Reply:
Sure.  Why not!